docs/DafnyRef/Types.md

```diff
@@ -1,18 +1,22 @@
 # 6. Types {#sec-types}
 ````grammar
-Type = DomainType [ "->" Type ]
+Type = DomainType_ | FunctionType_
 ````
 A Dafny type is a domain type (i.e., a type that can be the domain of a
 function type) optionally followed by an arrow and a range type.
 
 ````grammar
-DomainType =
+DomainType_ =
   ( BoolType_ | CharType_ | IntType_ | RealType_
   | OrdinalType_ | BitVectorType_ | ObjectType_
-  | FiniteSetType_ | InfiniteSetType_ | MultisetType_
-  | SequenceType_ | StringType_
-  | FiniteMapType_ | InfiniteMapType_ | ArrayType_
-  | TupleType_ | NamedType_
+  | FiniteSetType_ | InfiniteSetType_
+  | MultisetType_
+  | FiniteMapType_ | InfiniteMapType_
+  | SequenceType_
+  | StringType_
```

Comment:
If `string` is part of this list, then `nat` probably should be, too. Both are keywords and pre-defined types (`string` is a build-in type synonym, and `nat` is a built-in subset type).

Reply:
OK